.github/workflows/main.yml

```diff
@@ -50,7 +53,34 @@ jobs:
           flake8 . --exit-zero --max-complexity=6
 
       - name: Upload python artifacts
+        if: ${{ github.ref_name == 'main' }}
         uses: actions/upload-artifact@v4
         with:
           name: python-artifacts
           path: .
+
+  docker-ci:
+   if: ${{ github.ref_name == 'main' }}
+   runs-on: ubuntu-latest
+   needs: python-ci
+
+   steps:
+    - name: Download python artifacts
+      uses: actions/download-artifact@v4
+      if: ${{ github.ref_name == 'main' }}
```

Comment:
Recommend minimizing to use if conditions

Reply:
You have a condition that checks if the branch equals `main` [ 60 line ](https://github.com/mate-academy/devops_todolist_cicd_task_3_docker_ci/blob/f28805622be0450348989e98c09c3eab7d67c876/.github/workflows/main.yml#L60) So this check should be deleted